tests/test_config.py

```diff
@@ -173,3 +173,29 @@ def bob_config(transient_to, initial_config):
     assert "left" not in bob.text()
     assert "parted" not in bob.text()
     assert "quit" not in bob.text()
+
+
+# Neither Mantatail nor Hircd currently supports the WHOIS command.
+# IrcCore.nickmask will therefore always be "" in the test.
```

Comment:
We could also capture it when you join a channel, which will work most of the time. The `WHOIS` thing can still be useful. For example, if Alice is angry, and wants to write a long rant to Bob in a private message, she probably doesn't want to join any public channels if she can speak to Bob directly.

Reply:
I'm not completely sure what the point of this comment is. It was 9 days ago to be fair :)
What exactly is it you want to capture when joining a channel?

Reply:
When you join a channel, the IRC server sends you the user mask that we can then use, even if the server doesn't support WHOIS.